src/mediacatch_s2t/uploader.py

```diff
@@ -401,6 +400,7 @@ def upload_file(self):
 
 
 def upload_and_get_transcription(file, api_key, language):
+    update_myself()
```

Comment:
When the function is executed and the package is updated, the current run will still use the not-updated version since it is already loaded in RAM. The updated version will be used the next time the function is called.

Therefore, should we put the update_myself in the end of the function flow, just to ensure that no funky business is happening?

Reply:
interesting question. 
what is the multi-threaded scenario in here? A client worker that uses this package runs multithreading in the same environment?
If you would elaborate on the possible cases, I will test this package in those scenarios. For now, this package hasn't been test in multi-threading - single env.


> Has this been tested in a multi-threaded scenario?
> 
> E.g. what happens if the package is updated from one thread while the function is called from another? Is it atomic?

Reply:
Yes, it is multi-threaded single environment which I'm worried about.
Not sure if any of our customers has that setup, but just in case it should work.

Reply:
No, it's not atomic. The best approach for a user is to stop the running process before updating the package
> E.g. what happens if the package is updated from one thread while the function is called from another?
Is it atomic?

The update_myself method has been moved to the end of the function flow. But since it's not atomic, I don't think it will guarantee that it won't cause an error in a multi-threaded single env.
> Therefore, should we put the update_myself in the end of the function flow, just to ensure that no funky business is happening?